review deploy pipeline for Travis for executables

KtorZ opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->

#68 

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

This make sure that:

- Checks & tests are run for releases
- Releases happens when we actually push a tag on the repository (git tag --sign vX.Y.Z && git push origin --tags)
- Releases only happens for the last stage called 'Executables' (it was global before that!)
- Releases on GitHub are made in a 'Draft' mode, which means they aren't visible by anyone but maintainers, who
  can make a few edits before officially publishing it.
# Comments

<!-- Additional comments or screenshots to attach if any -->



<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
-->

KtorZ commented on src/Cardano/Wallet/API/V1/Types.hs:
```diff
@@ -558,11 +560,41 @@ mkTransactionSignatureAsBase16 (Core.Signature txSig) =
+instance ToSchema AddressPoolGap where
+  declareNamedSchema = genericDeclareNamedSchema defaultSchemaOptions
+
+instance ToJSONKey AddressPoolGap
```
:thinking: What's that for ?

denisshevchenko replied:
I don't know, just copied it... :-) Removed.

KtorZ commented on src/Cardano/Wallet/API/V1/Types.hs:
```diff
@@ -558,11 +560,41 @@ mkTransactionSignatureAsBase16 (Core.Signature txSig) =
+
+instance ToHttpApiData AddressPoolGap where
+    toQueryParam = fromString . show . getAddressPoolGap
+
```
_[must fix]_

I think we need a smart constructor here. Let's enforce a gap between 10 and 100 (completely arbitrary, but we can change later). 
Then, let's use that smart constructor to define the Arbitrary and HttpApiData instances.

KtorZ commented on src/Cardano/Wallet/Kernel/Wallets.hs:
```diff
@@ -236,7 +236,7 @@ createHdWallet pw mnemonic spendingPassword assuranceLevel walletName = do
                   -> [PublicKey]
                   -- ^ External wallet's accounts public keys.
-                  -> Word
+                  -> Word8
```
_[must fix]_

Why not `AddressPoolGap` here ?

denisshevchenko replied:
Because `AddressPoolGap` is defined in `API/V1/Types`. I shouldn't use types from wallet layer in the `Kernel`, right? Or I'm wrong?

KtorZ replied:
You're right. But, what about defining `AddressPoolGap` in the kernel then :trollface:  ?

KtorZ commented on src/Cardano/Wallet/API/V1/Types.hs:
```diff
@@ -558,11 +559,13 @@ mkTransactionSignatureAsBase16 (Core.Signature txSig) =
 instance Buildable [PublicKey] where
     build = bprint listJson
 
+deriveJSON Aeson.defaultOptions ''AddressPoolGap
```
_must fix_

Oversight here: this generates orphan instances, should be in the address gap module :+1:

KtorZ committed:
review deploy pipeline for Travis for executables

This make sure that:

- Checks & tests are run for releases
- Releases happens when we actually push a tag on the repository (git tag --sign vX.Y.Z && git push origin --tags)
- Releases only happens for the last stage called 'Executables' (it was global before that!)
- Releases on GitHub are made in a 'Draft' mode, which means they aren't visible by anyone but maintainers, who
  can make a few edits before officially publishing it.

KtorZ commented on .travis.yml:
```diff
@@ -86,6 +87,8 @@ jobs:
     script:
     - tar xzf $STACK_WORK_CACHE
+    - travis_retry curl -L -o hermes-testnet.tar.gz https://github.com/input-output-hk/cardano-wallet/raw/data-integration-testing/hermes-testnet.tar.gz
+    - tar xzf hermes-testnet.tar.gz -C $HOME
```
Whoops. This shouldn't be here :man_shrugging:

paweljakubas reviewed (approved):
Looks like a proper release policies inscribed to travis
